spec/models/manageiq/providers/google/network_manager/refresh_parser_spec.rb

```diff
@@ -0,0 +1,186 @@
+require 'digest'
+
+describe ManageIQ::Providers::Google::NetworkManager::RefreshParser do
+  let(:ems) { FactoryGirl.create(:ems_google, :project => "manageiq-dev") }
+  let(:az) { FactoryGirl.create(:availability_zone_google) }
+  let(:vm) do
+    FactoryGirl.create(:vm_google,
+                       :ext_management_system => ems,
+                       :ems_ref               => 123,
+                       :availability_zone     => az)
+  end
+  # These tests use some fairly aggressive stubbing. Specifically, all of the
+  # fog calls are stubbed out. This allows us to test the refresh logic without
+  # resorting to VCR casettes.
+  let(:connection) { double }
+
+  before(:example) do
+    allow(ems).to receive(:connect) { connection }
+    # Install some reasonable defaults, with all collections being empty.
+    allow(connection).to receive(:project) { ems.project }
+    allow(connection).to receive(:networks) { fog_collection([]) }
+    allow(connection).to receive(:firewalls) { fog_collection([]) }
+    allow(connection).to receive(:servers) { fog_collection([]) }
+    allow(connection).to receive(:addresses) { fog_collection([]) }
+    allow(connection).to receive(:forwarding_rules) { fog_collection([]) }
+    allow(connection).to receive(:target_pools) { fog_collection([]) }
+  end
+
+  it 'returns properly-structured hash on empty project' do
+    hashes = described_class.new(ems).ems_inv_to_hashes
+
+    expect(hashes).to eql(
+      :cloud_networks             => [],
+      :floating_ips               => [],
+      :load_balancer_listeners    => [],
+      :load_balancers             => [],
+      :load_balancer_pools        => [],
+      :load_balancer_pool_members => [],
+      :network_ports              => [],
+      :security_groups            => []
+    )
+  end
+
+  it 'returns a load balancer listener from a forwarding rule' do
```

Comment:
I'm having difficulty seeing what you're testing here. I think `ems_inv_to_hashes[:load_balancer]`. In that case, maybe:

```
describe 'load_balancers' do
  subject { described_class.new(ems).ems_inv_to_hashes[:load_balancers] } # Instead of 'hashes'
  before do
    # setup code
  end

  expect(subject).to eql()
end
```